subsys/net/lib/http/http_client.c

```diff
@@ -29,12 +29,28 @@ LOG_MODULE_REGISTER(net_http_client, CONFIG_NET_HTTP_LOG_LEVEL);
 #define HTTP_CONTENT_LEN_SIZE 11
 #define MAX_SEND_BUF_LEN 192
 
-static int sendall(int sock, const void *buf, size_t len)
+static int sendall(int sock, const void *buf, size_t len,
+					const k_timepoint_t req_end_timepoint)
 {
 	while (len) {
 		ssize_t out_len = zsock_send(sock, buf, len, 0);
 
-		if (out_len < 0) {
+		if ((out_len == 0) || (out_len < 0 && errno == EAGAIN)) {
+			struct zsock_pollfd pfd;
+			int pollres;
+			k_ticks_t req_timeout_ticks =
+				sys_timepoint_timeout(req_end_timepoint).ticks;
+			int req_timeout_ms = k_ticks_to_ms_floor32(req_timeout_ticks);
+
+			pfd.fd = sock;
+			pfd.events = ZSOCK_POLLOUT;
+			pollres = zsock_poll(&pfd, 1, req_timeout_ms);
+			if (pollres >= 0) {
```

Comment:
Shouldn't `pollres == 0` break the loop? Currently if we reach the timeout, it'd just start busy-looping.

Reply:
You are right. 
I'll correct it to 
```
pollres = zsock_poll(&pfd, 1, req_timeout_ms);
if (pollres == 0) {
    return -ETIMEDOUT;
}
else if (pollres > 0) {
    continue;
} else {
    return -errno;
}
```